### PR TITLE
deprecate willClone/didClone in favor of willCreateWorkingCopy/didCreateWorkingCopy

### PR DESCRIPTION
**Pull request by tomerd**
motivation: cloning has very specific meaning in git repository which is not quite what the delegate method represents, causing confusion

changes:
* deprecate willClone/didClone
* introduce willCreateWorkingCopy/didCreateWorkingCopy
* update call-sites and internal delegates
* adjust tests

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
This is great!  The wording here has always been confusing.  The only thing that I'd suggest changing is to say "copy" instead of "move", since it's not actually removing the clone.

**Comment by tomerd** [comment]
> This is great! The wording here has always been confusing. The only thing that I'd suggest changing is to say "copy" instead of "move", since it's not actually removing the clone.

@abertelrud went ahead and also renamed the APIs accordingly. PTAL

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
> > This is great! The wording here has always been confusing. The only thing that I'd suggest changing is to say "copy" instead of "move", since it's not actually removing the clone.
> 
> 
> 
> @abertelrud went ahead and also renamed the APIs accordingly. PTAL

Absolutely, I did mean to suggest using "copy" in the API itself, not just the comment.

This looks great to me now.  I found one comment that still possibly needs to be updated, but the rest looks good.

**Review by abertelrud — APPROVED** [review]

File: Sources/SourceControl/Repository.swift [hunk 1]
```diff
@@ -85,7 +85,7 @@ public protocol RepositoryProvider {
     /// - Throws: If the repository is unable to be opened.
     func open(repository: RepositorySpecifier, at path: AbsolutePath) throws -> Repository
 
-    /// Clone a managed repository into a working copy at on the local file system.
+    /// Create a working copy from a managed repository.
     ///
     /// Once complete, the repository can be opened using `openCheckout`. Note
```
Comment (tomerd) [inline Sources/SourceControl/Repository.swift:90]:
```suggestion
    /// Once complete, the repository can be opened using `openWorkingCopy`. Note
```

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**